planseqlearn/psl/mopa_mp_env.py

```diff
@@ -475,56 +450,168 @@ def get_object_pose_mp(self, **kwargs):
-            object_pos += np.array([-0.12, 0.05, 0.45])
-            if self.use_vision_pose_estimation:
+        if "hole" in obj_name:
+            if self.use_sam_segmentation:
+                object_pos = self.sam_object_pose[obj_name]
+            elif self.use_vision_pose_estimation:
                 object_pos = get_object_pose_from_seg(
                     self,
                     "4_part4_mesh",
                     "topview",
                     500,
                     500,
                     self._wrapped_env.sim,
-                    self,
-                    "4_part4_mesh",
-                    "topview",
-                    500,
-                    500,
-                    self._wrapped_env.sim,
                 )
-                object_pos += np.array([0, -0.3, 0.45])
+                object_pos += np.array([0., -0.3, 0.45])
+            else:
+                object_pos = get_site_pose(self._wrapped_env, "hole")[0]
+                object_pos += np.array([-0.12, 0.05, 0.45])
             object_quat = np.array([-0.50258679, -0.61890813, -0.49056324, 0.35172])
             object_quat /= np.linalg.norm(object_quat)
-        elif self.env_name == "SawyerPushObstacle-v0":
-            object_pos, object_quat = self.get_object_pose()
-            if self.use_vision_pose_estimation:
+        if "cube" in obj_name:
+            if self.use_sam_segmentation:
+                object_pos = self.sam_object_pose[obj_name] 
+            elif self.use_vision_pose_estimation:
                 object_pos = get_object_pose_from_seg(
                     self, "cube", "frontview", 500, 500, self._wrapped_env.sim
-                )
-            object_pos += np.array([-0.11, 0.045, 0.11])
-            object_quat = np.array([-0.57194288, 0.00869415, -0.77486997, 0.26903954])
+                ) 
+            else:
+                object_pos, _ = self.named_get_object_pose(obj_name)
+            object_pos += np.array([-0.14, 0.02, 0.06])
+            object_quat = None  
+        return object_pos, object_quat 
+    
+    def named_get_object_pose(self, obj_name):
+        if "can" in obj_name:
+            start = self.sim.model.body_jntadr[self.sim.model.body_name2id("cube")]
+            object_pos = self.sim.model.body_jntadr[self.sim.model.body_name2id("cube")]
+            object_pos = self.sim.data.qpos[start : start + 3].copy()
+            object_quat = self.sim.data.qpos[start + 3 : start + 7].copy()
+            object_quat = convert_quat(object_quat, to="xyzw")
+        if "hole" in obj_name:
+            object_pos = get_site_pose(self._wrapped_env, "hole")[0]
             object_quat = None
-        else:
-            raise NotImplementedError
-        return object_pos, object_quat
+        if "cube" in obj_name:
+            object_pos = np.array(
+                self._wrapped_env.sim.data.body_xpos[self._wrapped_env.cube_body_id]
+            )
+            object_quat = None
```

Comment:
why does this not have a sam option?

Reply:
this is just for get_object_pose which uses sim - get_object_pose_mp has the sam option

Reply:
can you update the naming convention so its very clear what the difference is between the two?